CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+    Position(int lat_deg, int lat_min, int lat_sec,
+             int lon_deg, int lon_min, int lon_sec)
+private:
+    double lat_{90.0};
+    double lon_{180.0};
+}
+
+// .cpp file
+Position::Position = default();
+Position::Position(int lat_deg, int lat_min, int lat_sec,
+                   int lon_deg, int lon_min, int lon_sec)
+: lat_{lms_to_deg(lat_deg, lat_min, lat_sec)}
+, lon_{lms_to_deg(lon_deg, lon_min, lon_sec)}
+{
+}
+```
+
+OKAY - Class member initialization is specified in one place:
+```c++
+// .h file
+class Position
+{
+public:
+    Position();
+    Position(int lat_deg, int lat_min, int lat_sec,
+             int lon_deg, int lon_min, int lon_sec);
+private:
+    double lat_;
+    double lon_;
+}
+
+// .cpp file
+Position::Position()
+: lat_{90.0}
+, lon_{180.0}
+{
+}
+Position::Position(int lat_deg, int lat_min, int lat_sec,
+                   int lon_deg, int lon_min, int lon_sec)
+: lat_{lms_to_deg(lat_deg, lat_min, lat_sec)}
+, lon_{lms_to_deg(lon_deg, lon_min, lon_sec)}
+{
+}
+
+```
+
+Zero-initialize fundamental member types with empty brackets, for
+consistency with default-initialized user-defined types.
+
+```c++
+class Navaid
+{
+public:
+    Navaid();
+private:
+    String<4> id_;
+    double lat_, lon_;
+    Frequency frequency_;
+}
+Navaid::Navaid()
+: id_{}         // --> default initialize string ("")
+, lat_{}        // --> 0.0
+, lon_{}        // --> 0.0
+, frequency_{}  // --> default initialize Frequency object
+{
+}
+```
+
+
+**Namespaces**
+
+Namespaces should be closed with a comment as in `} // namespace foos` or
+`} // namespace` for an unnamed (anonymous) namespace.
+
+Namespace contents (named or unnamed) should not be indented.
+
+**Namespace Using Directives**
+
+Namespace using directives (eg, `using namespace fplm;`) should not be used.
+
+**Enum Classes**
+
+The C++11 scoped `enum class` should be preferred over the unscoped C-style
+`enum`.
+
+**Null Values**
+
+The C++11 `nullptr` keyword should be used for the value of an uninitialized
+pointer instead of `0` or `NULL` value.  `'\0'` should be used for a null
+character.
+
+**Auto Keyword**
+
+Use of the `auto` keyword is encouraged for declaring local variables where
+it helps readability, but otherwise does not change the behavior of the code.
+
+Non-local variables should not be declared using `auto`, including function
+parameters or return types, except for creating a generic template-like
+function.
+
+**Assertions**
+
+Run-time conditional checks (assertions) should be used liberally to help
+expose software bugs in development and production builds.
+
+The standard `assert(cond)` macro from `<cassert>` prints to standard output and
+aborts the system if `cond` evaluates false.  When running on target hardware,
+standard output will not be available, so `assert(cond)` should not be used.
+
+Instead, a utility `ASSERT` macro should be created that is compatible with the
+target device, and it should be used in place of `assert(cond)`.
```

Comment:
I don't agree with this. I would rather use `assert(cond)` everywhere, and redefine `assert` if we need to.

I would rather see one
```c++
#ifndef NDEBUG
    #define assert(expr) \
      //stuff
#endif // NDEBUG
```

Rather than many
```c++
#ifndef NDEBUG
    ASSERT(cond);
#else
    assert(cond);
#endif // NDEBUG
```

The name doesn't matter to me (`ASSERT` is fine) as long as the same name is everywhere. I would also be OK with this:
```c++
#ifndef NDEBUG
    #define ASSERT(expr) \
      //stuff
#else
    #define ASSERT(expr) assert(expr)
#endif // NDEBUG
```
And then just use `ASSERT` everywhere.

Reply:
I just meant that we should have a special defined `ASSERT` that we use instead of the default.